django/db/models/sql/query.py

```diff
@@ -1928,8 +1928,17 @@ def set_group_by(self, allow_aliases=True):
         will be made automatically.
         """
         group_by = list(self.select)
-        if self.annotation_select:
-            for alias, annotation in self.annotation_select.items():
+        annotations_group_by = self.annotation_select.copy()
+        # Add not selected annotations used only in the ORDER BY clause.
+        if self.order_by and self.annotations:
+            for item in self.order_by:
+                if not hasattr(item, 'resolve_expression'):
```

Comment:
We allow for all kind of expressions in `order_by()`. I wonder if there an easy way to check annotations used in expressions :thinking: @charettes Any ideas?